### PR TITLE
Add class names to expose library elements for CSS styling

### LINKED ISSUE #1: Add class names to expose library elements for CSS styling
Add class names to the DOM elements in the `src/index.js` file.
A namespaced classname like `React_Virtualized_Listview__classname` is preferred.

Eg:
```jsx
<div style={style.container} ref={c => (this.container = c)} className="React_Virtualized_Listview__container">
```

### PR DESCRIPTION
**Pull request by rabingaire**
closes #1

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -87,7 +92,11 @@ export default class List extends React.Component {
   };
 
   renderList = () => (
-    <div style={style.container} ref={c => (this.container = c)}>
+    <div
+      style={style.container(this.getDefaultHeightWidth())}
+      className={this.props.className ? this.props.className : ''}
```
Comment (nishanbajracharya) [inline src/index.js]: We can just keep the class name as follows:
```js
className={this.props.className}
```
If the prop is not set, the value will be `undefined` and the className key will not be created. However if we use the ternary check, an empty classname will be set when classname prop is not passed.

File: src/index.js [hunk 2]
```diff
@@ -68,6 +67,10 @@ export default class List extends React.Component {
 
   getWrapper = () => ReactDOM.findDOMNode(this.listWrapper);
 
+  getDefaultHeightWidth = () => {
+    return this.props.className ? {} : { height: '300px', width: '600px' };
```
Comment (nishanbajracharya) [inline src/index.js]: Let's remove the redundant `return` and braces here.
```js
getDefaultHeightWidth = () => this.props.className ? {} : { height: '300px', width: '600px' };
```

Also why are we setting arbitrary values in the `height` and `width` props? Let's set them to `100%`. This way, the component height and width will default to using the full dimensions of its parent.

The user has to be able to use the component with as little configuration as possible. So setting a container width and height in the classname that they have to also set would be quite a lot of initial work.

File: src/index.js [hunk 3]
```diff
@@ -109,12 +118,13 @@ export default class List extends React.Component {
...
-  overScanCount: 5,
+  overScanCount: 5
 };
 
 List.propTypes = {
   renderItem: PropTypes.func,
   rowHeight: PropTypes.number,
   source: PropTypes.array.isRequired,
   overScanCount: PropTypes.number.isRequired,
+  className: PropTypes.string
```
Comment (nishanbajracharya) [inline src/index.js]: Length rule requirement here.

File: src/index.js [hunk 4]
```diff
@@ -3,18 +3,18 @@ import ReactDOM from 'react-dom';
...
 const style = {
-  container: {
-    width: '100%',
-    height: '100%',
-    position: 'relative'
-  },
+  container: ({ height, width }) => ({
+    position: 'relative',
+    height,
+    width
```
Comment (nishanbajracharya) [inline src/index.js]: Length rule requirement here.

**Review by nishanbajracharya — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 3 above)
Comment (rabingaire) [reply src/index.js]: I didn't get what you meant here?

File: src/index.js (hunk 4 above)
Comment (rabingaire) [reply src/index.js]: I didn't get what you meant here?

File: src/index.js (hunk 3 above)
Comment (nishanbajracharya) [reply src/index.js]: Order the proptypes based on the length of the line in ascending order. So the `className` prop needs to be third in this list. This is just a code formatting preference and doesn't affect the functionality of the code.

File: src/index.js (hunk 4 above)
Comment (nishanbajracharya) [reply src/index.js]: Same here.

**Review by nishanbajracharya — APPROVED** [review]

**Pull request merged**